src/webapp/components/data-submission/UploadRis.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { ChangeEvent, useEffect, useRef, useState } from "react";
+import { Button } from "@material-ui/core";
+import styled from "styled-components";
+import i18n from "@eyeseetea/d2-ui-components/locales";
+import BackupIcon from "@material-ui/icons/Backup";
+import CloseIcon from "@material-ui/icons/Close";
+
+interface UploadRisProps {
+    handleValidate: (val: boolean) => void;
+}
+export const UploadRis: React.FC<UploadRisProps> = ({ handleValidate }) => {
+    const [fileList, setFileList] = useState<FileList | null>(null);
+
+    useEffect(() => {
+        if (fileList && fileList?.length) {
+            handleValidate(true);
+        } else {
+            handleValidate(false);
+        }
+    }, [fileList, handleValidate]);
+
+    const inputRef = useRef<HTMLInputElement>(null);
+
+    const files = fileList ? [...fileList] : [];
+
+    const handleFileChange = (e: ChangeEvent<HTMLInputElement>) => {
+        setFileList(e.target.files);
+    };
+
+    const handleChooseFile = () => {
+        if (inputRef.current != null) {
+            inputRef.current.click();
+        }
+    };
+
+    const handleRemoveFiles = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
```

Comment:
[best practices, subjective] `handleXyz` are very common names, but using just the relevant verb is usually nicer. Here: `removeFiles`.

Reply:
follow-up on: https://app.clickup.com/t/860pp6vzv